ical/parsing/parser.py

```diff
@@ -24,97 +24,153 @@
+                            )
+                        else:
+                            raise CalendarParseError(
+                                f"Unexpected end of line. Expected ',', ';' or ':'",
+                                detailed_error=line,
+                            )
+
+                    if (char := line[pos]) == '"':
+                        if not quoted:
+                            raise CalendarParseError(
+                                f"Unexpected quote character outside parameter value",
+                                detailed_error=line,
+                            )
+
+                        param_value_read = True
+                        pos += 1
+
+                        if not (char := line[pos]) in (",", ";", ":"):
+                            raise CalendarParseError(
+                                f"Expected ',' or ';' or ':' after parameter value, got '{char}'",
+                                detailed_error=line,
+                            )
+
+                    elif not quoted and char in (",", ";", ":"):
+                        param_value_read = True
+
+                    if param_value_read:
+                        param_value = line[
+                            param_value_start : pos - (1 if quoted else 0)
+                        ]
+                        param_values.append(param_value)
+                        all_values_read = char != ","
+                        all_params_read = char == ":"
+
+                    pos += 1
+
+            params.append(ParsedPropertyParameter(name=param_name, values=param_values))
+
+            if not all_params_read:
+                # reset for next parameter
+                params_start = pos
+                pos += 1
+    else:
+        pos += 1
+
+    property_value = line[pos:]
+    if _RE_CONTROL_CHARS.search(property_value):
+        raise CalendarParseError(
+            f"Property value contains control characters: {property_value}",
+            detailed_error=line,
+        )
 
-_parser_lock = threading.Lock()
+    return ParsedProperty(
+        name=property_name, value=property_value, params=params if params else None
+    )
 
 
-def parse_contentlines(lines: Iterable[str]) -> list[ParseResults]:
-    """Parse a set of unfolded lines into parse results.
+def parse_contentlines(lines: Iterable[str]) -> list[ParsedProperty]:
+    """Parse a set of unfolded lines into parse results."""
 
-    Note, this method is not threadsafe and may be called from only one method at a time.
-    """
-    with _parser_lock:
-        parser = _create_parser()
-        try:
-            return [parser.parse_string(line, parse_all=True) for line in lines if line]
-        except ParseException as err:
-            raise CalendarParseError(
-                f"Failed to parse calendar contents", detailed_error=str(err)
-            ) from err
+    try:
+        return [parse_line(line) for line in lines if line]
+    except Exception as err:
```

Comment:
I didn't realize this was catching `Exception`.  This is catching an `IndexError` for one of the tests I added because it fails the check `if (char := line[pos]) == '"':` 

I think we're going to need to restructure this code to simplify this, taking a look.